extensions/amp-social-share/1.0/component.js

```diff
@@ -141,15 +141,34 @@ function checkProps(type, endpoint, target, width, height, params) {
     return null;
   }
 
+  const currentParams = Object.entries(typeConfig.defaultParams || {}).reduce(
```

Comment:
Would you mind adding a TODO linking to an issue describing the task to refactor this? Thanks!